data-raw/sample_data.R

```diff
@@ -109,46 +110,89 @@ sample_dfAnalyticsInput <- purrr::map(lAnalysis, "Analysis_Input") %>%
   dplyr::as_tibble() %>%
   dplyr::semi_join(site_subset)
 
-# Create User-facing Data ----
-
-## Before I was getting this lRaw at the same time as lMapped, but that caused me to
-## have duplicate columns during Analysis Data creation
-
-lUserWorkflows <- MakeWorkflowList(
-  strPath = system.file("workflow/userFacing", package = "gsm.app"),
-  strPackage = NULL
-)
-raw_spec <- CombineSpecs(lUserWorkflows)
-lRaw <- Ingest(lSource, raw_spec)
-lRaw <- lRaw %>%
-  purrr::map(dplyr::as_tibble)
-
-lUser <- RunWorkflows(lUserWorkflows, c(lRaw, lMapped))
-lUser <- lUser %>%
-  purrr::map(dplyr::as_tibble)
 
-# Prep domain data ----
+# lMapped for Domain Data ----
 
-lDomainData <- lUser
-names(lDomainData) <- stringr::str_remove(names(lDomainData), "^User_")
-lDomainData <- lDomainData[sort(names(lDomainData))]
+# SITE and STUDY are in dfGroups
+lMapped$Mapped_SITE <- NULL
+lMapped$Mapped_STUDY <- NULL
 
-# Apply subset to each df.
+# Apply subset to each mapped df.
 subject_groups <- dplyr::distinct(
   sample_dfAnalyticsInput,
   .data$SubjectID,
   .data$GroupID
 )
-lDomainData <- purrr::map(lDomainData, function(thisDomain) {
-  dplyr::inner_join(thisDomain, subject_groups, by = c("SubjectID"))
+sample_lMapped <- purrr::map(lMapped, function(thisDomain) {
+  dplyr::inner_join(
+    thisDomain,
+    subject_groups,
+    by = c("subjid" = "SubjectID")
+  )
 })
 
+# User-facing names ----
+
+# Ideally this should all be defined in gsm.mapping.
+
+chrDomainLabels <- c(
+  AE = "Adverse Events",
+  DATACHG = "Data Changes",
+  DATAENT = "Data Entry",
+  ENROLL = "Enrollment",
+  LB = "Lab",
+  PD = "Protocol Deviations",
+  QUERY = "Queries",
+  STUDCOMP = "Study Completion",
+  SUBJ = "Subject Metadata",
+  SDRGCOMP = "Treatment Completion"
+)
+
+chrFieldNames <- c(
```

Comment:
So, we want to keep these crosswalks out of the mapping yamls, because the preferred user-facing label might vary from module to module? or does it make sense to include a `label:` field in the mapping yamls if we want to standardize the user-facing labels across all modules? (i'm leaning towards the second idea, but open to comments)

Reply:
Ooooh, I love the idea of putting "label" in the yamls! I definitely intend for it to be attached to that field, so that makes sense!